libs/SalesforceAnalytics/src/com/salesforce/androidsdk/analytics/security/Encryptor.java

```diff
@@ -430,22 +472,33 @@ public static String decryptBytes(byte[] data, byte[] key, byte[] iv) {
 
     private static byte[] decrypt(byte[] data, int length, byte[] key, byte[] iv) throws GeneralSecurityException {
 
-        // Grabs the init vector prefix (first 16 bytes / 128 bits).
+        // Grabs the init vector prefix (first 12 bytes for GCM, or first 16 bytes for CBC).
         System.arraycopy(data, 0, iv, 0, iv.length);
 
         // Grabs the encrypted body after the init vector prefix.
         int meatLen = length - iv.length;
         int meatOffset = iv.length;
         byte[] meat = new byte[meatLen];
         System.arraycopy(data, meatOffset, meat, 0, meatLen);
-        final Cipher cipher = getDecryptingCipher(key, iv);
+        Cipher cipher;
+
+        // AES/GCM has an IV of length 12 bytes, whereas AES/CBC has an IV of length 16 bytes.
+        if (iv.length == 12) {
```

Comment:
These conditions will go away in `Mobile SDK 10.0` when we remove the upgrade steps.

Reply:
I don't think we need a `TODO:` here - I added a deprecation marker for `getLegacyDecryptingCipher`, so when we remove it in `10.0`, this will raise a compilation error and we will be forced to remove this then.